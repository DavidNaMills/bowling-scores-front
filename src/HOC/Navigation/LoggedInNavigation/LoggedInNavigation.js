import React from 'react';
import { NavLink } from 'react-router-dom';
import classes from '../Navigation.module.scss';

const LoggedInNavigation = (props) => {
    console.log(props.user);
    return (
        <div className={classes.navigation}>
            <div className={classes.navigation_section}>
                <NavLink exact className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/'>Home</NavLink>
                <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/game'>Game</NavLink>
            </div>
            <div className={classes.navigation_section}>
                <p className={classes.navigation__username} style={{color: `rgb(${props.user.color})`}}>{props.user.username}</p>
                <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/logout'>Logout</NavLink>
            </div>
        </div>
    )
}

export default LoggedInNavigation;