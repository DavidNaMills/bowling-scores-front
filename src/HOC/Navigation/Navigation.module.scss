@import '../../styles/colors.scss';

.navigation{
    background-color: $navigation_background ;
    padding: 5px;
    display: flex;
    justify-content: space-between;
    border-bottom: solid $popup_color 2px;

    &_section{
        // width: 120px;
        min-width: 120px;
        display:flex;
        justify-content: space-between;
    }

    &__link{
        color: $btn_font_clr;
        text-decoration: none;

        &:hover{
            color: rgb(28, 75, 135);
            font-weight: bold;
        }
        
        &_active{
            color: green;
            font-weight: bold;
        }
    }

    &__username{
        padding: 0;
        margin: 0;
        margin-right: 10px;
    }
}