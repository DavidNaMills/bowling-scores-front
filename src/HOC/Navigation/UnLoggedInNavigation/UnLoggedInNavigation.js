import React from 'react';
import { NavLink } from 'react-router-dom';
import classes from '../Navigation.module.scss';


const UnLoggedInNavigation = (props) => (
    <div className={classes.navigation}>
        <div className={classes.navigation_section}>
            <NavLink exact className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/'>Home</NavLink>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/game'>Game</NavLink>
        </div>
        <div className={classes.navigation_section}>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/login'>Login</NavLink>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/signup'>Signup</NavLink>
        </div>
    </div>
)

export default UnLoggedInNavigation;