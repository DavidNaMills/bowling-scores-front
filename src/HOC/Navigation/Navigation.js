import React from 'react';
import UnLoggedInNavigation from './UnLoggedInNavigation/UnLoggedInNavigation';
import LoggedInNavigation from './LoggedInNavigation/LoggedInNavigation';

const Navigation = (props) => {
    return (
        <header>
            <div>
                {
                    props.user.token
                        ? <LoggedInNavigation user={props.user.user} />
                        : <UnLoggedInNavigation />
                }
            </div>
            {
                props.children
            }
        </header>
    )
}

export default Navigation;