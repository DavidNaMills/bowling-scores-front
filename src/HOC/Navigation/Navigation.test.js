import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import Navigation from './Navigation';
import UnLoggedInNavigation from './UnLoggedInNavigation/UnLoggedInNavigation';
import LoggedInNavigation from './LoggedInNavigation/LoggedInNavigation';


const defaultUser = {
    token: null,
    user: null
}

const hasUser = {
    token: 'gdf5s6g5fd6sg5fd3s',
    user: {
        username: 'david'
    }
}

describe('<Navigation /> test suite', () => {
    describe('snapshot tests', () => {
        it('should match the snapshot <UnLoggedInNavigation/>', () => {
            const component = renderer.create(
                <BrowserRouter>
                    <Navigation user={defaultUser}/>
                </BrowserRouter>
            );
            const app = component.toJSON();
            expect(app).toMatchSnapshot();
        });

        it('should match the snapshot <LoggedInNavigation/>', () => {
            const component = renderer.create(
                <BrowserRouter>
                    <Navigation user={hasUser}/>
                </BrowserRouter>
            );
            const app = component.toJSON();
            expect(app).toMatchSnapshot();
        });
    });

    describe('render tests', () => {
        it('renders UnLoggedInNavigation if no user token is present', ()=>{
            const wrapper = mount(
                <BrowserRouter>
                    <Navigation user={defaultUser}/>
                </BrowserRouter>
            );

            expect(wrapper.find(UnLoggedInNavigation).length).toBe(1);
            expect(wrapper.find(LoggedInNavigation).length).toBe(0);
        });
        
        it('renders LoggedInNavigation when user token is present', ()=>{
            const wrapper = mount(
                <BrowserRouter>
                    <Navigation user={hasUser}/>
                </BrowserRouter>
            );

            expect(wrapper.find(UnLoggedInNavigation).length).toBe(0);
            expect(wrapper.find(LoggedInNavigation).length).toBe(1);
        });
    });
});

