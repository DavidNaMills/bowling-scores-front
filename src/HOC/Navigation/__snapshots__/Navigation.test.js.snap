// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> test suite snapshot tests should match the snapshot <LoggedInNavigation/> 1`] = `
<header>
  <div>
    <div
      className="navigation"
    >
      <div
        className="navigation_section"
      >
        <a
          aria-current="page"
          className="navigation__link navigation__link_active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="navigation__link"
          href="/game"
          onClick={[Function]}
        >
          Game
        </a>
      </div>
      <div
        className="navigation_section"
      >
        <p>
          david
        </p>
        <a
          aria-current={null}
          className="navigation__link"
          href="/logout"
          onClick={[Function]}
        >
          Logout
        </a>
      </div>
    </div>
  </div>
</header>
`;

exports[`<Navigation /> test suite snapshot tests should match the snapshot <UnLoggedInNavigation/> 1`] = `
<header>
  <div>
    <div
      className="navigation"
    >
      <div
        className="navigation_section"
      >
        <a
          aria-current="page"
          className="navigation__link navigation__link_active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="navigation__link"
          href="/game"
          onClick={[Function]}
        >
          Game
        </a>
      </div>
      <div
        className="navigation_section"
      >
        <a
          aria-current={null}
          className="navigation__link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          aria-current={null}
          className="navigation__link"
          href="/signup"
          onClick={[Function]}
        >
          Signup
        </a>
      </div>
    </div>
  </div>
</header>
`;
